app/controllers/api/v1/concerns/authenticator.rb

```diff
@@ -58,6 +62,11 @@ def get_current_api_user
                 user.update_last_access_at!
                 return user
               end
+            else
```

Comment:
It really feels like the cyclomatic complexity of this method is ballooning.  We might want to extract some logic into helper functions as it's already at 40 lines now.